### PR TITLE
[SYCL][NFC] Avoid multiple execution of expression passed to macro

### PR DESCRIPTION
**Pull request by vladimirlaz**
Signed-off-by: Vladimir Lazarev <vladimir.lazarev@intel.com>

### CODE REVIEW

**Comment by vladimirlaz** [comment]
> [NFC]?

done

File: sycl/include/CL/sycl/detail/common.hpp [hunk 1]
```diff
@@ -22,8 +22,10 @@
 
 const char *stringifyErrorCode(cl_int error);
 
-#define OCL_CODE_TO_STR(code)                                                  \
-  std::string(std::to_string(code) + " (" + stringifyErrorCode(code) + ")")
+static inline  std::string codeToString(cl_uint code){
```
Comment (alexeyvoronov) [inline sycl/include/CL/sycl/detail/common.hpp]:
```suggestion
static inline std::string codeToString(cl_int code){
```
Double space.

Why cl_uint? stringifyErrorCode expects cl_int.

File: sycl/include/CL/sycl/detail/common.hpp [hunk 2]
```diff
@@ -32,21 +34,27 @@ const char *stringifyErrorCode(cl_int error);
 
 #ifndef SYCL_SUPPRESS_OCL_ERROR_REPORT
 #include <iostream>
-#define REPORT_OCL_ERR_TO_STREAM(code)                                         \
+#define REPORT_OCL_ERR_TO_STREAM(expr)                                         \
```
Comment (alexeyvoronov) [inline sycl/include/CL/sycl/detail/common.hpp]: Let's change it to inline function also

File: sycl/include/CL/sycl/detail/common.hpp [hunk 3]
```diff
@@ -32,21 +34,27 @@ const char *stringifyErrorCode(cl_int error);
...
 
-#define REPORT_OCL_ERR_TO_EXC(code, exc)                                       \
+#define REPORT_OCL_ERR_TO_EXC(expr, exc)                                       \
+{                                                                              \
+  auto code = expr;                                                            \
   if (code != CL_SUCCESS) {                                                    \
-    std::string errorMessage(OCL_ERROR_REPORT + OCL_CODE_TO_STR(code));        \
+    std::string errorMessage(OCL_ERROR_REPORT + codeToString(code));           \
     std::cerr << errorMessage << std::endl;                                    \
     throw exc(errorMessage.c_str(), (code));                                   \
```
Comment (alexeyvoronov) [inline sycl/include/CL/sycl/detail/common.hpp]:
```suggestion
    throw exc(errorMessage, code);                                   \
```
`exc` has constructor from a string.
useless brackets

File: sycl/include/CL/sycl/detail/common.hpp (hunk 2 above)
Comment (vladimirlaz) [reply sycl/include/CL/sycl/detail/common.hpp]: I ma trying to minimize change.

**Comment by bader** [comment]
@vladimirlaz, please, resolve conflicts.

**Comment by vladimirlaz** [comment]
> @vladimirlaz, please, resolve conflicts.

done

**Pull request merged**